data_structures/binary_tree/segment_tree_other.py

```diff
@@ -4,17 +4,29 @@
 function takes 2 values and returns a same type value
 """
 from collections.abc import Sequence
+from functools import reduce
 from queue import Queue
+from typing import TypeVar
+
+T = TypeVar("T", int, float)
 
 
 class SegmentTreeNode:
-    def __init__(self, start, end, val, left=None, right=None):
+    def __init__(
+        self,
+        start: int,
+        end: int,
+        val: T,
+        left: "SegmentTreeNode" = None,
+        right: "SegmentTreeNode" = None,
+    ):
         self.start = start
         self.end = end
         self.val = val
         self.mid = (start + end) // 2
         self.left = left
         self.right = right
+        self.lazy: T = None
```

Comment:
Perhaps T should be renamed to `IntOrFloat`.  Given the definition of `T`, it is surprising that we are setting `self.lazy` to `None`.

Reply:
Sounds good, I will rename `T`.

Reply:
@cclauss Done.